pyperf/_runner.py

```diff
@@ -496,6 +496,43 @@ def task_func(task, loops):
         task.inner_loops = inner_loops
         return self._main(task)
 
+    def bench_async_func(self, name, func, *args, **kwargs):
+        """Benchmark await func(*args)"""
+
+        inner_loops = kwargs.pop('inner_loops', None)
+        metadata = kwargs.pop('metadata', None)
+        self._no_keyword_argument(kwargs)
+
+        if not self._check_worker_task():
+            return None
+
+        if args:
+            func = functools.partial(func, *args)
+
+        def task_func(task, loops):
+            import asyncio
+            # use fast local variables
+            local_timer = time.perf_counter
+            local_func = func
+            loop = asyncio.get_event_loop()
```

Comment:
Would it make sense to attempt reimplementing asyncio.run() here, but only measure run_until_complete() timing? I'm talking about creating a fresh event loop each time, and shutdown asynchronous generators and the default executor. Well, same things than asyncio.run().

Reply:
I use asyncio.run() instead of reimplement it.